src/filesystem.cpp

```diff
@@ -58,8 +58,12 @@ bool do_mkdir(std::string const& path, int const mode)
 
 bool assure_dir_exist(std::string const &path)
 {
-    std::unique_ptr<DIR, decltype(&closedir)> dir {opendir(path.c_str()), closedir};
-    return dir || do_mkdir(path, 0777);
+  DIR *dir=opendir(path.c_str());
+  if(dir!=NULL){
+    closedir(dir);
+    return true;
+  }
+  return dir || do_mkdir(path, 0777);
```

Comment:
Indentation should be 4 spaces per level, not 2. Also needs some whitespaces added  - see doc/CODE_STYLE.txt. There is a stray space added to the Makefile, please remove it.

The code is fine, but checking `dir` here is not needed. It will always be null, otherwise the if block above would have left the function already.

 And while I'm nitpicking: prefer using `nullptr` instead of `NULL` as it is more type safe.